src/OpenTelemetry.AutoInstrumentation.Native/CMakeLists.txt

```diff
@@ -96,26 +94,40 @@ SET(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${OUTPUT_BIN_DIR})
 # Dependencies
 # ******************************************************
 
-# Add custom commands to setup dependencies via git
-add_custom_command(
-    OUTPUT ${OUTPUT_DEPS_DIR}/json/include/nlohmann/json.hpp
-    COMMAND git clone -c advice.detachedHead=false --quiet --depth 1 --branch v3.11.2 https://github.com/nlohmann/json.git
-    WORKING_DIRECTORY ${OUTPUT_DEPS_DIR}
-)
+include(FetchContent)
 
-add_custom_command(
-    OUTPUT ${OUTPUT_DEPS_DIR}/re2/obj/libre2.a
-    COMMAND git clone -c advice.detachedHead=false --quiet --depth 1 --branch 2022-12-01 https://github.com/google/re2.git && cd re2 && env ARFLAGS=\"-r -s -c\" CXXFLAGS=\"-std=c++11 -O3 -g -fPIC\" make
-    WORKING_DIRECTORY ${OUTPUT_DEPS_DIR}
+# Fetch json library
+FetchContent_Declare(
+    json
+    GIT_REPOSITORY https://github.com/nlohmann/json.git
+    GIT_TAG v3.11.2
```

Comment:
I removed referenced to nlohman/json in #2637.
While working on this, please apply this changes here.